src/RJP.MultiUrlPicker/Models/MultiUrls.cs

```diff
@@ -39,12 +40,14 @@ public MultiUrls(string propertyData)
 
         private void Initialize(JArray data)
         {
+            var umbracoHelper = new Lazy<UmbracoHelper>(() => new UmbracoHelper(UmbracoContext.Current));
```

Comment:
By using `Lazy<UmbracoHelper>` and passing the same instance, it is only created when actually needed and only once (if multiple links are added).